### PR TITLE
fix: `ThemeResource` bindings are not scope-updated on `Style` switch

### LINKED ISSUE #12496: [Overrides][PipsPager][Fluent] Overrides on the style of PipsPager do not work well
### Current behavior

image
image
When overriding the style of the pipspager with static resources, as you can see in the first image, the selected pipspager has the correct colors while the other's didnt change. When you change the selection, none of the colors follow the override anymore. 

### Expected behavior

image
It should be like on WinUI where the colors override are done, and stay overridden. 

### How to reproduce it (as minimally and precisely as possible)

Put that into the code 
```
<Page.Resources>
        <SolidColorBrush x:Key="PipsPagerSelectionIndicatorBackgroundSelected" Color="Red" />
        <SolidColorBrush x:Key="PipsPagerSelectionIndicatorForegroundSelected" Color="Pink" />
        <SolidColorBrush x:Key="PipsPagerSelectionIndicatorForeground" Color="Purple" />
        <SolidColorBrush x:Key="PipsPagerSelectionIndicatorBackground" Color="Blue" />

    </Page.Resources>


    <StackPanel HorizontalAlignment="Center"
				VerticalAlignment="Center">
        <PipsPager/>
    </StackPanel>
```
### Workaround

NA

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

For WASM
`<PackageReference Include="Uno.Wasm.Bootstrap" Version="8.0.0-dev.210" />`
`<PackageReference Include="Uno.Wasm.Bootstrap.DevServer" Version="8.0.0-dev.210" />`
`<PackageReference Include="Uno.WinUI.WebAssembly" Version="4.10.0-dev.111" />`
`<PackageReference Include="Uno.WinUI.RemoteControl" Version="4.10.0-dev.111"/> `
`<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />`

On android: `<PackageReference Include="Uno.WinUI" Version="4.10.0-dev.111" />`

### Affected platforms

WebAssembly, Android, Skia (WPF), Skia (GTK on Linux/macOS/Windows)

### IDE

Visual Studio 2022

### IDE version

Version 17.6.0

### Relevant plugins

_No response_

### Anything else we need to know?

I will try on IOS later but it might take a while

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #12496 

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Previously only `StaticResource` bindings were re-evaluated when `Style` was applied. This is not enough as `ThemeResources` also need to be sourced from the current scope. In addition, the precedence override needs to be applied only while the setters are being applied, but not for the resource bindings update, otherwise it could also apply base control bindings with `ExplicitStyle` precedence, even though the bindings have just `ImplicitStyle` precedence.

## What is the new behavior?

Fixed!

### <samp>🤖 Generated by Copilot at e51e120</samp>

This pull request fixes a bug and adds a test for theme resource resolution in styles. It refactors the `Style.ApplyTo` method to make it more clear and consistent. It also creates a new user control with two styles for a button, one using a theme resource and one using a static resource, and a new test method that asserts that the button's background color changes correctly when the style is switched at runtime.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Style/Style.cs [hunk 1]
```diff
@@ -91,15 +91,17 @@ internal void ApplyTo(DependencyObject o, DependencyPropertyValuePrecedences pre
 					}
 				}
 
+				localPrecedenceDisposable?.Dispose();
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Style/Style.cs:94]: Are we disposing twice now?

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Style/Style.cs:94]: We can do the refactor for 5x after https://github.com/unoplatform/uno/issues/11977

**Review by Youssef1313 — APPROVED** [review]

**Review by agneszitte — APPROVED** [review]

**Review by Xiaoy312 — APPROVED** [review]

**Comment by Xiaoy312** [comment]
this should be a backport candidate for 4x

**Comment by MartinZikmund** [comment]
@Xiaoy312 yes, backport will follow up after it is merged 👍

**Comment by agneszitte** [comment]
formatting issue @MartinZikmund 
**src\Uno.UI\UI\Xaml\Style\Style.cs(86,1): Error IDE0055: Fix formatting**

**Comment by MartinZikmund** [comment]
@mergifyio backport legacy/4x

**Pull request merged**